packages/langium/src/parser/cst-node-builder.ts

```diff
@@ -89,9 +89,9 @@ export class CstNodeBuilder {
 
     construct(item: { $type: string | symbol | undefined, $cstNode: CstNode }): void {
         const current: CstNode = this.current;
-        // The specified item could be a datatype ($type is symbol) or a fragment ($type is undefined)
+        // The specified item could be a datatype ($type is symbol), fragment ($type is undefined) or infix rule ($infix exists)
         // Only if the $type is a string, we actually assign the element
-        if (typeof item.$type === 'string') {
+        if (typeof item.$type === 'string' && !('$infix' in item)) {
```

Comment:
Wrt. performance: Would it be advantageous to add `$infix?: boolean` to the type def of `item` and just test `!item.$infix`?